sdk/src/wallet/core/operations/storage.rs

```diff
@@ -36,9 +36,7 @@ mod storage_stub {
     {
         async fn save(&self, storage: &impl StorageAdapter<Error = crate::wallet::Error>) -> crate::wallet::Result<()> {
             log::debug!("save_wallet_data");
-            storage
-                .set(WALLET_INDEXATION_KEY, &WalletBuilderDto::from(self))
-                .await?;
+            storage.set(WALLET_INDEXATION_KEY, self).await?;
```

Comment:
How does it know to use the DTO ?

Reply:
I made the impls one-sided, so you can serialize the builder but have to deserialize the dto.